hooks/test_post_gen_project.py

```diff
@@ -48,3 +48,15 @@ def test_copy_journal_template_files_bad(
         match=errormessage,
     ):
         copy_journal_template_files(input, project_dir)
+
+
+# C1: an existing gh repo url.
+#   Expect return the cloned path.
+def test_clone_gh_repo():
+    pass
+
+
+# C1: an not existing gh repo url.
+#   Expect FileNotFoundError.
```

Comment:
Here are some different fails I can think of:
1. an improperly formed url . expect ValueError
2. a properly formed url but GH repo cannot be found.  Expect pass through the error from the request
3. GH repo can be found but some other cloning error. Expect pass through the error from the request

That's all I can think of, because if it clones successfully, then the Path will be generated.

Another note.  This task is already being done in `scikit-package` I think, or in cookiecutter, or probably somehere else on the internet.  We can consider reusing a function for doing this from somewhere else.  In that case we don't need any tests and we don't need any function.  So it is worth looking for that first.

Reply:
> I found a module `gitpython`

great!  First check it is a good package.  Is it well maintained or does it look like it was written by a grad student (!) :).  If it looks like a professionally maintained package we can make it a dependency and use its function that does what we want to do.  We may not need to write this function at all, or we may want our function to be a thin wrapper of that function that raises more helpful error messages for our needs.

The next step then is for you to look into that package and propose a way forward, justifying your decisions a bit (you don't have to write too much....use Bob as kind of inspiration.  He is a very clear communicator.